aten/src/ATen/core/ivalue.h

```diff
@@ -502,6 +502,10 @@ struct CAFFE2_API IValue final {
 
   TypePtr type() const;
 
+  size_t hash() const {
+    return payload.as_int;
```

Comment:
Why are we doing this ?

Reply:
To track aliased attributes. For now the JIT is not preserving aliasing (except for tensors) but in future the current logic can deal with: 
self.a = [1,2]
self.b = self.a